agent/agent_endpoint_test.go

```diff
@@ -4551,7 +4551,7 @@ func TestAgentConnectCARoots_list(t *testing.T) {
 
 	// Set some CAs. Note that NewTestAgent already bootstraps one CA so this just
 	// adds a second and makes it active.
-	ca2 := connect.TestCAConfigSet(t, a, nil)
+	ca2 := connect.TestCAConfigSet(t, a, nil, connect.DefaultPrivateKeyType, connect.DefaultPrivateKeyBits)
```

Comment:
Rather than adding this boiler plate everywhere, I'd suggest we leave the old `TestCAConfigSet`signature alone but create a new one `TestCAConfigSetWithKeyType` with this new signature and then redefine `TestCAConfigSet` to call `TestCAConfigSetWithKeyType` with the default params.

I appreciate how much effort you put into finding and replacing all these but I think the simpler change is worth it for the code base in the long run - adding more params is inevitable but we have strong precedent for leaving the basic case simple and only opting in to the more complex boiler plate in tests where variation is needed.

Reply:
Same for `TestCAWithKeyType` etc.

Reply:
Fixed with 17eb913659f179f8996fa1416d858eb1fb1d1de6.